finality-aleph/src/sync/handler.rs

```diff
@@ -0,0 +1,486 @@
+use crate::{
+    session::{last_block_of_session, session_id_from_block_num, SessionId, SessionPeriod},
+    sync::{
+        data::{NetworkData, State},
+        forest::{Error as ForestError, Forest, JustificationAddResult},
+        BlockIdFor, BlockIdentifier, ChainStatus, Finalizer, Header, Justification, PeerId,
+        Verifier,
+    },
+};
+
+/// How many justifications we will send at most in response to a broadcast.
+const MAX_SMALL_JUSTIFICATION_BATCH: u32 = 10;
+// Silly workaround to make range matching actually work...
+const MAX_SMALL_JUSTIFICATION_BATCH_PLUS_ONE: u32 = MAX_SMALL_JUSTIFICATION_BATCH + 1;
```

Comment:
really?...

Reply:
Oh believe me, I'm very open to improvement suggestions. Dx

Reply:
I mean, maybe don't make this a const, but rather create a local variable right before the match?

Reply:
oh, it does not work...

Reply:
ok, I'm not sure if this qualifies as "improvement", but I'd suggest:
1. nested `match`, outer branch matches `Some(n)`
2. checked `n -= 1`
3. use non inclusive ranges: `0`, `1..MAX`, and `MAX..`

Reply:
I mean it's ugly, but we don't have to worry that someone forgets to update the other const in the future, and since that particular number was chosen pseudorandomly, then it might be necessary to tune it later...

Reply:
hmm, then again, the compiler will complain

Reply:
it's fine, closing...